server/routes/summary_routes.ts

```diff
@@ -57,33 +55,26 @@ export function registerSummaryAssistantRoutes(
         topNLogPatternData: req.body.topNLogPatternData,
       });
       let summary;
-      let insightAgentIdExists = false;
+      let insightAgentId;
       try {
         if (req.body.insightType) {
           // We have separate agent for os_insight and user_insight. And for user_insight, we can
           // only get it by searching on name since it is not stored in agent config.
           if (req.body.insightType === 'os_insight') {
-            if (!osInsightAgentId) {
-              osInsightAgentId = await getAgentIdByConfigName(OS_INSIGHT_AGENT_CONFIG_ID, client);
-            }
-            insightAgentIdExists = !!osInsightAgentId;
-          } else if (req.body.insightType === 'user_insight') {
-            if (req.body.type === 'alerts') {
-              if (!userInsightAgentId) {
-                userInsightAgentId = await searchAgent({ name: 'KB_For_Alert_Insight' }, client);
-              }
-            }
-            insightAgentIdExists = !!userInsightAgentId;
+            insightAgentId = await getAgentIdByConfigName(OS_INSIGHT_AGENT_CONFIG_ID, client);
+          } else if (req.body.insightType === 'user_insight' && req.body.type === 'alerts') {
+            insightAgentId = await searchAgent({ name: 'KB_For_Alert_Insight' }, client);
```

Comment:
Shall we store opensearch kb agent id and customer kb agent id into two separate variables? I wonder whether there is some override issue.

Reply:
It doesn't store the agent id but only pass it to front-end

Reply:
Makes sense